util/info.c

```diff
@@ -331,6 +349,7 @@ int main(int argc, char **argv)
 	hints->mode = ~0;
 	hints->domain_attr->mode = ~0;
 	hints->domain_attr->mr_mode = ~(FI_MR_BASIC | FI_MR_SCALABLE);
+	free(hints->fabric_attr->prov_name);
```

Comment:
fi_freeinfo() will free the name if it is set.  This should result in a double free.